src/components/search/search.stories.mdx

```diff
@@ -27,154 +28,36 @@ import Search from "carbon-react/lib/components/search";
 
-### Search without Search Button and default width
+### Default
 
 <Canvas>
-  <Story name="default" parameters={{ info: { disable: true } }}>
-    <Search placeholder="Search..." defaultValue="" />
-  </Story>
+  <Story name="default" story={stories.Default} />
 </Canvas>
 
-### Search can be controlled
+### Controlled
 
 <Canvas>
-  <Story name="controlled" parameters={{ info: { disable: true } }}>
-    {() => {
-      const [value, setValue] = useState("");
-      return (
-        <div>
-          <Button onClick={() => setValue("")}>Clear value</Button>
-          <Button onClick={() => setValue("test value")} ml={2}>
-            Set value
-          </Button>
-          <Search
-            id="test"
-            name="test"
-            placeholder="Search..."
-            onChange={(e) => setValue(e.target.value)}
-            value={value}
-            searchButton
-          />
-        </div>
-      );
-    }}
-  </Story>
+  <Story name="controlled" parameters={{ info: { disable: true } }} story={stories.Controlled} />
 </Canvas>
 
-### Search with Search Button and default width
+### With Search Button
 
 <Canvas>
-  <Story name="with SearchButton" parameters={{ info: { disable: true } }}>
-    <Search defaultValue="Here is some text" searchButton />
-  </Story>
+  <Story name="with SearchButton" parameters={{ info: { disable: true } }} story={stories.WithSearchButton} />
 </Canvas>
 
-### Search without Search Button, default width and colour background
-
-<Canvas style={{ backgroundColor: "#e6ebed" }}>
-  <Story
-    name="default with colour background"
-    parameters={{ info: { disable: true }, chromatic: { disable: true } }}
-  >
-    <Search placeholder="Search..." defaultValue="" />
-  </Story>
-</Canvas>
-
-### Search with Search Button, default width and colour background
-
-<Canvas style={{ backgroundColor: "#e6ebed" }}>
-  <Story
-    name="with SearchButton and colour background"
-    parameters={{ info: { disable: true }, chromatic: { disable: true } }}
-  >
-    <Search
-      placeholder="Search..."
-      defaultValue="Here is some text"
-      searchButton
-    />
-  </Story>
-</Canvas>
-
-### Search without Search Button and custom width
+### With Custom width
 
 In this example the `searchWidth` prop is 375px.
 
 <Canvas>
-  <Story name="custom Width using px" parameters={{ info: { disable: true } }}>
-    <Search placeholder="Search..." defaultValue="" searchWidth="375px" />
-  </Story>
+  <Story name="with custom width" parameters={{ info: { disable: true } }} story={stories.WithCustomWidth} />
 </Canvas>
 
-### Search with Search Button and custom width
-
-In this example the `searchWidth` prop is 375px.
+### Dark Variant
 
 <Canvas>
-  <Story
-    name="with SearchButton and Custom Width using px"
-    parameters={{ info: { disable: true } }}
-  >
-    <Search defaultValue="Here is some text" searchButton searchWidth="375px" />
-  </Story>
-</Canvas>
-
-### Search without Search Button and custom width
```

Comment:
Again, is there a good reason to remove all these stories you've removed?

Reply:
I think that we don't have to show every prop combination to our users, that makes the documentation over bloated. I think that was done for testing purposes. All of those combinations could be tested in the test story.